### PR TITLE
Replaced kyma/epp package with local EPP package and fixed lint issues

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Replaced kyma/epp package with local EPP package.
- Fixed lint issues.

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/18439

### CODE REVIEW

File: pkg/cloudevents/eventtype/eventtypetest/eventtypetest.go [hunk 1]
```diff
@@ -15,7 +15,3 @@ func (c CleanerStub) Clean(_ string) (string, error) {
 func (cf CleanerFunc) Clean(eventType string) (string, error) {
 	return cf(eventType)
 }
-
-var DefaultCleaner = func(eventType string) (string, error) {
```
Comment (friedrichwilken) [inline pkg/cloudevents/eventtype/eventtypetest/eventtypetest.go:19]: I guess it was not used?

File: pkg/nats/connect.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 
 type Opt = nats.Option
 
+//nolint:gochecknoglobals // cloning functions as variables.
 var (
```
Comment (friedrichwilken) [inline pkg/nats/connect.go:12]: Could we store them as consts? Never really tried that but I am wondering.

File: pkg/oauth/client.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ func NewClient(ctx context.Context, cfg *env.EventMeshConfig) *http.Client {
 	// configure tracing transport
 	client.Transport = &ochttp.Transport{
 		Base:        client.Transport,
-		Propagation: tracecontextb3.TraceContextEgress,
+		Propagation: tracecontextb3.TraceContextEgress(),
```
Comment (friedrichwilken) [inline pkg/oauth/client.go:29]: That was a litte hard to understand. How did you catch that? Did it really work before?

File: pkg/cloudevents/eventtype/eventtypetest/eventtypetest.go (hunk 1 above)
Comment (mfaizanse) [reply pkg/cloudevents/eventtype/eventtypetest/eventtypetest.go:19]: yes, it was not used.

File: pkg/nats/connect.go (hunk 2 above)
Comment (mfaizanse) [reply pkg/nats/connect.go:12]: I tried, but it gives error that `nats.RetryOnFailedConnect` is not a const.

File: pkg/oauth/client.go (hunk 3 above)
Comment (mfaizanse) [reply pkg/oauth/client.go:29]: It was a global variable, I tried to change it to const but it did not work, so I changed it to a function.

**Review by friedrichwilken — APPROVED** [review]

**Pull request merged**